src/tyr/route_serializer_osrm.cc

```diff
@@ -1197,11 +1170,187 @@ json::MapPtr osrm_maneuver(const valhalla::DirectionsLeg::Maneuver& maneuver,
       }
     }
   }
+  return maneuver_type;
+}
+
+// Populate the OSRM maneuver record within a step.
+json::MapPtr osrm_maneuver(const valhalla::DirectionsLeg::Maneuver& maneuver,
+                           const std::string& maneuver_type,
+                           const std::string& modifier,
+                           const uint32_t in_brg,
+                           const uint32_t out_brg,
+                           const PointLL& man_ll,
+                           const bool emplace_instructions) {
+  auto osrm_man = json::map({});
+
+  // Set the location
+  auto loc = json::array({});
+  loc->emplace_back(json::fixed_t{man_ll.lng(), 6});
+  loc->emplace_back(json::fixed_t{man_ll.lat(), 6});
+  osrm_man->emplace("location", loc);
+
+  osrm_man->emplace("bearing_before", static_cast<uint64_t>(in_brg));
+  osrm_man->emplace("bearing_after", static_cast<uint64_t>(out_brg));
   osrm_man->emplace("type", maneuver_type);
 
+  if (emplace_instructions) {
+    osrm_man->emplace("instruction", maneuver.text_instruction());
+  }
+  if (!modifier.empty()) {
+    osrm_man->emplace("modifier", modifier);
+  }
+  // Roundabout count
+  if (maneuver.type() == DirectionsLeg_Maneuver_Type_kRoundaboutEnter &&
```

Comment:
ok so this gets pulled out of the guts of the maneuver function to the top level

Reply:
This was previously nested within the `osrm_maneuver` part where the `maneuver_type` was determined. `maneuver_type` had to become its own function, so it could be reused in the `banner_instructions` though. This was the only part that affected the `osrm_man`, so it had to be a part of the `osrm_maneuver` function.